schema/schema_validator.py

```diff
@@ -298,28 +320,28 @@ def check_test_output_schema(self, icu_version, test_type, executor):
         return results
 
     def validate_schema_file(self, schema_file_path):
+        test_type = os.path.basename(os.path.dirname(schema_file_path))
```

Comment:
We should not infer `test_type` based on the filename. The filename could be subject to change. What seems more authoritative is the contents in the file itself. In the JSON structure that you get after parsing, you can look at `properties` -> `test_type` -> `enum` -> 0th element. That seems more definitive and less error-prone.

Reply:
Agreed. This is a good idea.

Reply:
I'm now getting the test type from the file contents. That required adding this to all the files and updating the schema. It's a good change!